internal/tiltfile/value/path.go

```diff
@@ -24,3 +26,53 @@ func (p *LocalPath) Unpack(v starlark.Value) error {
 	p.Value = str
 	return nil
 }
+
+type LocalPathList struct {
+	t                  *starlark.Thread
+	Value              []string
+	AllowsSingleString bool
+}
+
+func NewLocalPathListUnpacker(t *starlark.Thread) LocalPathList {
+	return LocalPathList{
+		t: t,
+	}
+}
+
+func (p *LocalPathList) Unpack(v starlark.Value) error {
+	if p.AllowsSingleString {
+		_, ok := AsString(v)
+		if ok {
+			str, err := ValueToAbsPath(p.t, v)
+			if err != nil {
+				return err
+			}
+			p.Value = []string{str}
+			return nil
+		}
+	}
+
+	var iter starlark.Iterator
+	switch x := v.(type) {
+	case *starlark.List:
+		iter = x.Iterate()
+	case starlark.Tuple:
+		iter = x.Iterate()
+	default:
+		return fmt.Errorf("value should be a string or List or Tuple of strings, but is of type %s", v.Type())
```

Comment:
"string or" should only be included if `p.AllowsSingleString`

Alternatively, I'd be in favor of deleting that option and always behaving as if it's true. AFAIK the only reason we are inconsistent on whether a single string is allowed is a lack of attention paid to consistency (and the fact that we previously had this logic duplicated everywhere and no centralized place to standardize the behavior). Maybe I'm missing something or maybe changing the existing API is more scope than you want in this PR, though.

Reply:
ok, i buy that argument!